homeassistant/components/launch_library/sensor.py

```diff
@@ -1,69 +1,75 @@
-"""A sensor platform that give you information about the next space launch."""
+"""Support for Launch Library sensors."""
 from __future__ import annotations
 
-from datetime import timedelta
-import logging
-
-from pylaunches import PyLaunches, PyLaunchesException
-import voluptuous as vol
-
-from homeassistant.components.sensor import PLATFORM_SCHEMA, SensorEntity
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME
-from homeassistant.helpers.aiohttp_client import async_get_clientsession
-import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from .const import (
     ATTR_AGENCY,
     ATTR_AGENCY_COUNTRY_CODE,
     ATTR_LAUNCH_TIME,
     ATTR_STREAM,
     ATTRIBUTION,
-    DEFAULT_NAME,
+    DOMAIN,
 )
 
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(hours=1)
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string}
-)
+    coordinator = hass.data[DOMAIN]
 
+    sensors = [
+        NextLaunchSensor(coordinator, "Next launch"),
+    ]
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
+    async_add_entities(sensors, True)
 
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
 
+class LLBaseEntity(CoordinatorEntity, SensorEntity):
+    """Sensor base entity."""
 
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
+    def __init__(self, coordinator: DataUpdateCoordinator, name: str) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
+        self._attr_name = name
+        self._attr_unique_id = f"{DOMAIN}/{name}"
```

Comment:
I would recommend using a key for the unique ID, e.g., `next_launch` instead of the user-friendly name.

Reply:
Do you mean like this instead?

```
_attr_unique_id = f"{DOMAIN}/next_launch"
```

Reply:
Yes, or `f"{DOMAIN}_next_launch"`

Reply:
The unique ID is pr. domain so that is not needed to add here?

Reply:
Right that is even better 👍

Reply:
I am not sure i understand what you mean 🤔

Reply:
@ludeeus Do you mean that i should just remove `DOMAIN` and just do this instead `f"next_launch"`?

Reply:
It would be just `_attr_unique_id = "next_launch"` but that is conflicting with the rules of unique ID's
So let's do `_attr_unique_id = f"{coordinator.config_entry.entry_id}_next_launch"` instead

Reply:
`mypy` complains when trying to access `entry_id`. Is this because we dump the coordinator directly into `hass.data[DOMAIN]` like this `hass.data[DOMAIN] = coordinator`?
```
homeassistant/components/launch_library/sensor.py:65: error: Item "None" of "Optional[ConfigEntry]" has no attribute "entry_id"  [union-attr]
```

Reply:
Right...
Since we don't have a custom coordinator here, just pass the entry_id in the constructor instead

Reply:
I believe i have resolved this now